wasmcloud/tests/smtp_integration_test.rs

```diff
@@ -0,0 +1,457 @@
+use std::net::IpAddr;
+use std::time::Duration;
+
+use futures::StreamExt;
+use serde_json::{Value, json};
+use serial_test::serial;
+use testcontainers::core::logs::LogFrame;
+use testcontainers::runners::AsyncRunner;
+use testcontainers::{ContainerAsync, ImageExt};
+use testcontainers::{CopyDataSource, GenericImage, core::IntoContainerPort, core::WaitFor};
+use tokio::process::Command;
+use tokio::sync::OnceCell;
+use tokio::time::sleep;
+use wadm_client::{ClientConnectOptions, ManifestLoader};
+
+const NATS_PORT: u16 = 4222;
+const MAILCATCHER_SMTP_PORT: u16 = 25;
+const MAILCATCHER_API_PORT: u16 = 80;
+const SMTP_COMPONENT_PORT: u16 = 8000;
+
+const WASMCLOUD_VERSION: &str = "1.8.0";
+const WADM_VERSION: &str = "v0.15.0-wolfi";
+const NATS_VERSION: &str = "2.10.19-alpine";
+
+type ContainerDef = ContainerAsync<GenericImage>;
+
+static ONCES: OnceCell<(ContainerDef, ContainerDef, ContainerDef, ContainerDef)> =
+    OnceCell::const_new();
+
+// NOTE: Keeping it here for debugging purposes
+fn show_logs(logframe: &LogFrame) {
+    match logframe {
+        LogFrame::StdOut(bytes) => println!("{}", String::from_utf8_lossy(bytes)),
+        LogFrame::StdErr(bytes) => println!("{}", String::from_utf8_lossy(bytes)),
+    }
+}
+
+async fn build_wasm() {
+    let output = Command::new("wash")
+        .arg("build")
+        .current_dir("../helper/smtp-provider")
+        .output()
+        .await
+        .unwrap();
+
+    if !output.status.success() {
+        println!(
+            "Failed building provider, Stdout: {}, stderr: {}",
+            String::from_utf8_lossy(&output.stdout),
+            String::from_utf8_lossy(&output.stderr)
+        );
+        panic!()
+    }
+
+    let output = Command::new("wash")
+        .args(vec!["wit", "deps"])
```

Comment:
wash wit deps is being called by wash build (you can test by deleting the deps folder of the component)

Reply:
weird